Several flaky tests in  io.camunda.zeebe.it.client.command.AssignUserTaskTest

**Summary**

- How often does the test fail? Once
- Does it block your work? No 
- Do we suspect that it is a real failure?

**Failures**

shouldRejectIfUserTaskIsAlreadyAssignedWithProhibitedOverride: io.camunda.zeebe.it.client.command.AssignUserTaskTest
```
[Expected a created user task] 
Expecting actual:
  -1L
to be greater than:
  0L
java.lang.AssertionError: 
[Expected a created user task]
Expecting actual:
-1L
to be greater than:
0L


at io.camunda.zeebe.it.util.ZeebeResourcesHelper.createSingleUserTask(ZeebeResourcesHelper.java:187)
at io.camunda.zeebe.it.util.ZeebeResourcesHelper.createSingleUserTask(ZeebeResourcesHelper.java:173)
at io.camunda.zeebe.it.client.command.AssignUserTaskTest.initClientAndInstances(AssignUserTaskTest.java:42)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
```

shouldRejectIfMissingAssignee: io.camunda.zeebe.it.client.command.AssignUserTaskTest
```
java.lang.AssertionError: 
[Expected a created user task] 
Expecting actual:
  -1L
to be greater than:
  0L
at io.camunda.zeebe.it.util.ZeebeResourcesHelper.createSingleUserTask(ZeebeResourcesHelper.java:187)
at io.camunda.zeebe.it.util.ZeebeResourcesHelper.createSingleUserTask(ZeebeResourcesHelper.java:173)
at io.camunda.zeebe.it.client.command.AssignUserTaskTest.initClientAndInstances(AssignUserTaskTest.java:42)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
```
**Hypotheses**

None

**Logs**
https://github.com/camunda/zeebe/actions/runs/8786220325

[Test results for [IT] QA Integration Tests.zip](https://github.com/camunda/zeebe/files/15072345/Test.results.for.IT.QA.Integration.Tests.zip)

--- a/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/AssignUserTaskTest.java
+++ b/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/AssignUserTaskTest.java
@@ -27,17 +27,16 @@ import org.junit.jupiter.api.Test;
 @AutoCloseResources
 class AssignUserTaskTest {
 
-  @TestZeebe
-  private static final TestStandaloneBroker ZEEBE =
-      new TestStandaloneBroker().withRecordingExporter(true);
-
   @AutoCloseResource ZeebeClient client;
+
+  @TestZeebe
+  private final TestStandaloneBroker zeebe = new TestStandaloneBroker().withRecordingExporter(true);
 
   private long userTaskKey;
 
   @BeforeEach
   void initClientAndInstances() {
-    client = ZEEBE.newClientBuilder().defaultRequestTimeout(Duration.ofSeconds(15)).build();
+    client = zeebe.newClientBuilder().defaultRequestTimeout(Duration.ofSeconds(15)).build();
     final ZeebeResourcesHelper resourcesHelper = new ZeebeResourcesHelper(client);
     userTaskKey = resourcesHelper.createSingleUserTask();
   }

